app/controllers/campaign_pages_controller.rb

```diff
@@ -1,2 +1,32 @@
 class CampaignPagesController < ApplicationController
+
+  def new
+    @campaign_page = CampaignPage.new
+    @templates = Template.where active: true
+  end
+
+  def create
+    permitted_params = CampaignPageParameters.new(params).permit
+    if permitted_params[:slug].nil?
+      permitted_params[:slug] = permitted_params[:title].parameterize
+    end
+    permitted_params[:active] = true
+    permitted_params[:featured] = false
+    permitted_params[:language_id] = 1
+    page = CampaignPage.create! permitted_params
+    widgets = params[:widgets]
+    i = 0
```

Comment:
Maybe a comment here to explain what we're doing with the widgets? @Tuuleh what do you think?